src/msalexample-ios-macos/Shared/MSALAuthentication.swift

```diff
@@ -0,0 +1,80 @@
+import MSAL
+
+class MSALAuthentication {
+    // 'Application (client) ID' of app registration in Azure portal - this value is a GUID
+    private static let kClientId = ""
+
+    // 'Tenant ID' of your Azure AD instance - this value is a GUID
+    private static let kTenantId = ""
+    
+    private static let kAuthority = try! MSALB2CAuthority(url: URL(string: "https://login.microsoftonline.com/\(kTenantId)")!)
+    private static let kConfig = MSALPublicClientApplicationConfig(clientId: kClientId, redirectUri: nil, authority: kAuthority)
+
+    // In order to take advantage of token caching, your MSAL client singleton must
+    // have a lifecycle that at least matches the lifecycle of the user's session in
+    // the app.
+    private static let kMSALClient: MSALPublicClientApplication = try! MSALPublicClientApplication(configuration: kConfig)
+    
+    public static func signin(completion: @escaping (_ accessToken: String?) -> Void) {
+        #if os(iOS)
+        let scenes = UIApplication.shared.connectedScenes
+        let windowScene = scenes.first as? UIWindowScene
+        let window = windowScene?.windows.first
+        // IMPORTANT: For this sample it is possible to use the root, consider discovering the top one
+        // or passing a specific ViewController if required.
+        let webviewParameters = MSALWebviewParameters(authPresentationViewController: window!.rootViewController!)
+        #else
+        let webviewParameters = MSALWebviewParameters()
+        #endif
+
+        let interactiveParameters = MSALInteractiveTokenParameters(scopes: ["user.read"], webviewParameters: webviewParameters)
+
+        // If access token acquisition needs to happen multiple times in
+        // the app, only call this after checking for a cached token via
+        // a call to kApplication.acquireTokenSilent(with: MSALSilentTokenParameters).
```

Comment:
```suggestion
        // a call to kDrawRestOfOwl(with: MSALSilentTokenParameters).
```

:stuck_out_tongue_winking_eye:

Reply:
happy to validate this as a learning @mmacy. We are now adding the silent acquisition over here as well.

done  | addressed from 2bb9a46c5943ddcf778cd8b3f276e4057102114d